src/main/java/com/example/demo/service/impl/UserServiceImpl.java

```diff
@@ -55,6 +57,9 @@ public boolean isOrderOwner(
             final Integer userId,
             final Integer orderId
     ) {
-        return userRepository.isOrderOwner(userId, orderId);
+        if(orderRepository.findById(orderId).get().getUser().getId() == userId){
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return orderRepository.findById(orderId).get().getUser().getId() == userId;
```

а почему мы сравниваем ссылочные типы через двойное равно? конечно в integer есть кэш и если там будут значения между -128 и 127 то проблем мы не получим но все равно... да и нет смысла тут в параметрах именно ссылочные типы использовать, можно примитивные